lib/rubygems/gemcutter_utilities/webauthn_listener.rb

```diff
@@ -85,10 +85,17 @@ def root_path?(uri)
     end
 
     def parse_otp_from_uri(uri)
-      require "cgi"
+      query = uri.query
+      return unless query && !query.empty?
 
-      return if uri.query.nil?
-      CGI.parse(uri.query).dig("code", 0)
+      query.split('&').each do |param|
```

Comment:
`.each` is not needed.
```suggestion
      query.split('&') do |param|
```